justfile

```diff
@@ -0,0 +1,57 @@
+[doc('Build all targets in the project')]
+build-all:
+  bazel build //...
```

Comment:
My 2c, we shouldn't have any `build //*/...` commands in here because:
1. They'll build a lot of intermediate targets that may not be necessary for checking if the code works
2. Not all targets are buildable by themselves. Some of them require the context of what they're being built for, i.e. `cc_library` compilation changes depending on whether it's for JVM or Android. If there is no default set up, likely because is doesn't make sense to build that target by itself, then this would fail.

These scripts should be intent based, "what are you building for" -- testing, publishing, executing.

Reply:
I agree with the general sentiment, but I'm not sure I'd go so far to say we shouldn't have _any_ `//...` commands here. I don't expect them to be used very often, but for things like testing, we may want to test the full project. 

> Not all targets are buildable by themselves

I'll admit I'm not as familiar with the standard patterns for `cc_library` -- but if that truly is the case, then I'd expect we should exclude those targets from the `build-all` target. Maybe it's more beneficial to convert this into a macro for building all of the sub-systems: 

```
build-core
build-plugins
build-android
build-react
build-ios
```

which I assume is similar to what we'd want to do for `lint-all`, `test-all`, etc

Reply:
> I'm not sure I'd go so far to say we shouldn't have any //... commands here. I don't expect them to be used very often, but for things like testing

Using `...` expansion for testing is fine, but that's because tests explicitly declare how the dependencies should be built. But I'm sticking to avoiding it for `build` commands -- you probably don't need to build the targets used for publishing locally, at least at the `build-all` level. It's somewhat misleading to expect all the tooling targets to also be build with a `build-all` command.

Reply:
This also isn't limited to just the `cc_library` targets - any Bazel target may be transitioned upon to change the underlying toolchain. It's called out in the iOS docs for the Swift-based targets:
https://github.com/player-ui/player/tree/main/ios#bazel